### DISCUSSION TITLE
instance_constraints

### DISCUSSION THREAD

**Discussion by Peter230655** [csu-hmc/opty#212] [Q&A]
As I understand the documentation, the instance constraints must be numbers, they cannot be intervals.
Is my understanding correct?
If correct, would there be an easy way to change this?
I could imagine optimizations, where the final target need not be hit exactly, but a range would be good enough.
(I had such a simulatiom one time, where it was impossible to hit a point, but a range would have been possible)
**Thanks!**

**Comment by moorepants** [comment]
>  I could imagine optimizations, where the final target need not be hit exactly, but a range would be good enough.

A range should be specified as an inequality constraint, which is not yet supported.

**Reply by moorepants** [reply]
>  but a range would be good enough

Do you mean a range in time? Or a range of values at a specific time?

**Comment by Peter230655** [comment]
I meant a range in values, not in time.

**Reply by moorepants** [reply]
A range in values can be supported by the inequality constraints, which you've been discussing in other issues.

**Reply by Peter230655** [reply]
> A range in values can be supported by the inequality constraints, which you've been discussing in other issues.

Yes, there is an open issue.
I only thought, that if the inequalities only concern the final time (at time tf $x(tf) \in (a, b)$ this might be easier.

**Reply by moorepants** [reply]
If you look at how IPOPT is designed, there is only one way to state any given equation is a an equality or inequality. So the solution to inequalities, in general, can also be used for instance constraints.
